browser/ui/views/permissions/permission_prompt_bubble_base_view_browsertest.cc

```diff
@@ -0,0 +1,112 @@
+#include "testing/gtest/include/gtest/gtest.h"
+
+using PermissionPromptBubbleBaseViewBrowserTest = InProcessBrowserTest;
+
+namespace {
+
+class MockPermissionPromptDelegate
+    : public permissions::PermissionPrompt::Delegate {
+ public:
+  MockPermissionPromptDelegate() {
+    request_.push_back(std::make_unique<permissions::PermissionRequest>(
+        std::make_unique<permissions::PermissionRequestData>(
+            std::make_unique<permissions::ContentSettingPermissionResolver>(
+                ContentSettingsType::MEDIASTREAM_CAMERA),
+            true, GURL()),
+        base::DoNothing()));
+  }
+  ~MockPermissionPromptDelegate() override = default;
+
+  const std::vector<std::unique_ptr<permissions::PermissionRequest>>& Requests()
+      override {
+    return request_;
+  }
+
+  GURL GetRequestingOrigin() const override { return GURL(); }
+
+  GURL GetEmbeddingOrigin() const override { return GURL(); }
+
+  void Accept() override {}
+  void AcceptThisTime() override {}
+  void Deny() override {}
+  void Dismiss() override {}
+  void Ignore() override {}
+  void FinalizeCurrentRequests() override {}
+  void OpenHelpCenterLink(const ui::Event& event) override {}
+  void PreIgnoreQuietPrompt() override {}
+  void SetManageClicked() override {}
+  void SetLearnMoreClicked() override {}
+  void SetHatsShownCallback(base::OnceCallback<void()> callback) override {}
+
+  bool WasCurrentRequestAlreadyDisplayed() override { return false; }
+  bool ShouldDropCurrentRequestIfCannotShowQuietly() const override {
+    return false;
+  }
+  bool ShouldCurrentRequestUseQuietUI() const override { return false; }
+  std::optional<permissions::PermissionUiSelector::QuietUiReason>
+  ReasonForUsingQuietUi() const override {
+    return std::nullopt;
+  }
+  void SetDismissOnTabClose() override {}
+  void SetPromptShown() override {}
+  void SetDecisionTime() override {}
+  bool RecreateView() override { return false; }
+  const permissions::PermissionPrompt* GetCurrentPrompt() const override {
+    return nullptr;
+  }
+
+  base::WeakPtr<permissions::PermissionPrompt::Delegate> GetWeakPtr() override {
+    return weak_ptr_factory_.GetWeakPtr();
+  }
+
+  content::WebContents* GetAssociatedWebContents() override { return nullptr; }
+
+ private:
+  std::vector<std::unique_ptr<permissions::PermissionRequest>> request_;
+
+  base::WeakPtrFactory<MockPermissionPromptDelegate> weak_ptr_factory_{this};
+};
+
+class MockPermissionPromptBubbleBaseView
+    : public PermissionPromptBubbleBaseView {
+ public:
+  MockPermissionPromptBubbleBaseView(
+      Browser* browser,
+      base::WeakPtr<permissions::PermissionPrompt::Delegate> delegate)
+      : PermissionPromptBubbleBaseView(browser,
+                                       delegate,
+                                       base::TimeTicks::Now(),
+                                       PermissionPromptStyle::kBubbleOnly) {
+    CreateWidget();
+    ShowWidget();
```

Comment:
I didn't notice this during code review, so I'm curious about what is going on here. Where would the definition of these functions be?

Reply:
Yes, just calling the base's methods. `CreateWidget` is protected member, so I made this class.